statsmodels/tsa/statespace/mlemodel.py

```diff
@@ -282,6 +300,150 @@ def loglike(self, params=None, average_loglike=False, transformed=True,
+        increments = np.identity(n) * 1j * epsilon
+
+        # Get values at the params themselves
+        self.update(params)
+        res = self.filter(**kwargs)
+        dtype = self.dtype
+        # Save this for inversion later
+        inv_forecasts_error_cov = res.forecasts_error_cov.copy()
+
+        # Compute partial derivatives
+        partials_forecasts_error = (
+            np.zeros((self.k_endog, self.nobs, n))
+        )
+        partials_forecasts_error_cov = (
+            np.zeros((self.k_endog, self.k_endog, self.nobs, n))
+        )
+        for i, ih in enumerate(increments):
+            self.update(params + ih)
+            res = self.filter(**kwargs)
+
+            partials_forecasts_error[:, :, i] = (
+                res.forecasts_error.imag / epsilon[i]
+            )
+
+            partials_forecasts_error_cov[:, :, :, i] = (
+                res.forecasts_error_cov.imag / epsilon[i]
+            )
+
+        # Compute the information matrix
+        tmp = np.zeros((self.k_endog, self.k_endog, self.nobs, n), dtype=dtype)
+
+        information_matrix = np.zeros((n, n), dtype=dtype)
+        for t in range(self.loglikelihood_burn, self.nobs):
+            inv_forecasts_error_cov[:, :, t] = (
+                np.linalg.inv(inv_forecasts_error_cov[:, :, t])
+            )
+            for i in range(n):
+                tmp[:, :, t, i] = np.dot(
+                    inv_forecasts_error_cov[:, :, t],
+                    partials_forecasts_error_cov[:, :, t, i]
+                )
+            for i in range(n):
+                for j in range(n):
+                    information_matrix[i, j] += (
+                        0.5 * np.trace(np.dot(tmp[:, :, t, i],
+                                              tmp[:, :, t, j]))
+                    )
+                    information_matrix[i, j] += np.inner(
+                        partials_forecasts_error[:, t, i],
+                        np.dot(inv_forecasts_error_cov[:,:,t],
+                               partials_forecasts_error[:, t, j])
+                    )
+        return information_matrix
+
+
+    def opg_information_matrix(self, params, **kwargs):
+        """
+        Outer product of gradients information matrix
+
+        Parameters
+        ----------
+        params : array_like, optional
+            Array of parameters at which to evaluate the loglikelihood
+            function.
+        **kwargs
+            Additional arguments to the `loglikeobs` method.
+
+        References
+        ----------
+        Berndt, Ernst R., Bronwyn Hall, Robert Hall, and Jerry Hausman. 1974.
+        Estimation and Inference in Nonlinear Structural Models.
+        NBER Chapters. National Bureau of Economic Research, Inc.
+
+        """
+        self.update(params)
+        scoreobs = approx_fprime_cs(params, self.loglikeobs,
+                                    kwargs=kwargs).transpose()
+        return np.inner(scoreobs, scoreobs)
```

Comment:
scoreobs should be a separate method, since those are also used for the sandwiches